File: concrete-sample/src/androidTest/java/com/jaynewstrom/concretesample/MainActivityTest.java
package com.jaynewstrom.concretesample;

import android.support.test.rule.ActivityTestRule;
import android.support.test.runner.AndroidJUnit4;

import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.assertion.ViewAssertions.matches;
import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
import static android.support.test.espresso.matcher.ViewMatchers.withText;

@RunWith(AndroidJUnit4.class)
public final class MainActivityTest {

    @Rule public ActivityTestRule<MainActivity> activityRule = new ActivityTestRule<>(MainActivity.class);

    @Test public void ensureActivityIsStarted() {
        onView(withText("Hello World!")).check(matches(isDisplayed()));
    }
}

File: concrete-sample/src/androidTest/java/com/jaynewstrom/concretesample/main/MainActivityTest.kt
package com.jaynewstrom.concretesample.main

import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
import androidx.test.espresso.matcher.ViewMatchers.withId
import androidx.test.rule.ActivityTestRule
import androidx.test.runner.AndroidJUnit4
import com.jaynewstrom.concretesample.R
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class MainActivityTest {
    @get:Rule var activityRule = ActivityTestRule(MainActivity::class.java)

    @Test fun ensureActivityIsStarted() {
        onView(withId(R.id.first_details_button)).check(matches(isDisplayed()))
    }
}
